pyomo/contrib/gdpopt/tests/test_gdpopt.py

```diff
@@ -826,27 +1195,93 @@ def test_RIC_custom_disjuncts(self):
              eight_process.use_unit_8ornot.disjuncts[0]]
         ]
 
-        def assert_correct_disjuncts_active(nlp_model, solve_data):
-            if solve_data.master_iteration >= 1:
+        def assert_correct_disjuncts_active(subprob_util_block,
+                                            master_util_block):
+            # I can get the iteration based on the number of no-good
+            # cuts in this case...
+            iteration = len(master_util_block.no_good_cuts)
```

Comment:
This is a good point. I just made a change so that all the callbacks accept the solver object as an argument, and then you can get into whatever trouble you want... :)